tiledb/sm/query/query.cc

```diff
@@ -1529,15 +1206,24 @@ Status Query::set_validity_buffer(
         "Cannot set buffer; " + name + " validity buffer size is null"));
   }
 
-  // Must be an attribute
-  if (!array_schema_->is_attr(name)) {
+  // Must be an attribute or aggregate
+  const bool is_aggr = is_aggregate(name);
+  if (!array_schema_->is_attr(name) && !is_aggr) {
     return logger_->status(Status_QueryError(
         std::string("Cannot set buffer; Buffer name '") + name +
         "' is not an attribute"));
   }
 
   // Must be nullable
-  if (!array_schema_->is_nullable(name)) {
+  bool is_nullable;
+  if (is_aggr) {
+    is_nullable = array_schema_->is_nullable(
+        default_channel_aggregates_[name]->field_name());
```

Comment:
I haven't gotten to the implementation of `default_channel_aggregates_`, but I'm super confused on what this logic is doing without that context. I *think* this is saying "if the aggregate is nullable", but I then have a bunch more questions on what a nullable aggregate means. I.e., is this a "aggregates aggregate non-null values" vs a "aggregates encountring a null value become null because what does 1 + null mean other than null" situation?

Reply:
And its just occurred to me that "default" is applied to the "channel" not "aggregates". I.e., we're not adding a default "count" aggregate, we're checking the "default channel" for aggregates. I'm... not sure about the channels analogy just yet.

Reply:
Some aggregates might output a NULL value. For example if you have a SUM aggregate and all the values are NULL, I think that the output of the SUM should be NULL versus 0. For MIN/MAX, this will also make sense...

The default channel is the output of a regular query today. All the aggregates supported in this PR are aggregating on the default channel and output only one value. Later the user will have the ability to create more channels that might have GROUP BY, for which we'll need aggregators that can aggregate on multiple values.